service/src/main/java/com/bankbazaar/kafka/service/config/KafkaStreamsConfig.java

```diff
@@ -0,0 +1,151 @@
+import lombok.extern.slf4j.Slf4j;
+import org.apache.kafka.streams.kstream.KStream;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.cloud.stream.annotation.StreamRetryTemplate;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.retry.RetryPolicy;
+import org.springframework.retry.backoff.FixedBackOffPolicy;
+import org.springframework.retry.policy.SimpleRetryPolicy;
+import org.springframework.retry.support.RetryTemplate;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+@Slf4j
+@Configuration
+public class KafkaStreamsConfig {
+    @Autowired
+    private FileStatusService fileStatusService;
+
+    @Value("${spring.datasource.maxRetries}")
+    private Integer maxRetries;
+
+    /**
+     *Setup retry template bean.
+     */
+    @Bean
+    @StreamRetryTemplate
+    RetryTemplate streamRetryTemplate() {
+        RetryTemplate retryTemplate = new RetryTemplate();
+        RetryPolicy retryPolicy = new SimpleRetryPolicy(maxRetries);
+        FixedBackOffPolicy backOffPolicy = new FixedBackOffPolicy();
+        backOffPolicy.setBackOffPeriod(1);
+        retryTemplate.setBackOffPolicy(backOffPolicy);
+        retryTemplate.setRetryPolicy(retryPolicy);
+        return retryTemplate;
+    }
+    /**
+     * Set status to IN_PROGRESS.
+     * Consumes data from Create_CSV topic.
+     * Validates if file already exits.
+     * Set status to FAILURE if file already exists.
+     * Output data to File_Processor topic.
+     */
+    @Bean
+    public Function<KStream<String,Data>, KStream<String,Data>> fileProcessor(@Qualifier("streamRetryTemplate") RetryTemplate retryTemplate)
+    {
+        return kStream -> kStream.filter((key, value) ->
+                {
+                    return retryTemplate.execute(
+                            retryContext -> {
+                                try {
+                                    FileStatusEntity fileData = new FileStatusEntity();
+                                    fileData.setId(value.getId());
+                                    fileData.setStatus(Status.IN_PROGRESS);
+                                    fileStatusService.update(fileData);
+                                    ClassLoader classLoader = getClass().getClassLoader();
+                                    File file = new File(classLoader.getResource(".").getFile() + value.getFileName());
+                                    if (file.exists()) {
+                                        fileData.setStatus(Status.FAILURE);
+                                        fileStatusService.update(fileData);
+                                        return false;
```

Comment:
This method is supposed to return of type Data and you are returning a boolean.

Reply:
this method now return Data type